app/src/pages/Editor/Document.pup.spec.ts

```diff
@@ -0,0 +1,137 @@
+/** @jest-environment ./src/util/jest_puppeteer_electron_environment */
+
+import { RootState } from '../../state';
+
+jest.setTimeout(60000);
+
+import { JSONObject } from 'puppeteer-core';
+import { DocumentItem, Document } from '../../core/document';
+import { assertSome } from '../../util';
+
+const testContent: DocumentItem[] = [
+  { type: 'heading', level: 1, text: 'First big heading' },
+  { type: 'paragraph_break', speaker: 'Speaker One' },
+  { type: 'word', source: 'source-1', sourceStart: 2, length: 1, word: 'One', conf: 1 },
+  { type: 'word', source: 'source-1', sourceStart: 3, length: 1, word: 'Two', conf: 1 },
+  { type: 'word', source: 'source-1', sourceStart: 4, length: 1, word: 'Three', conf: 1 },
+  { type: 'word', source: 'source-1', sourceStart: 5, length: 1, word: 'Four', conf: 1 },
+  { type: 'paragraph_break', speaker: null },
+  { type: 'heading', text: 'Heading Two', level: 2 },
+  { type: 'paragraph_break', speaker: 'Speaker Two' },
+  { type: 'word', source: 'source-2', sourceStart: 2, length: 1, word: 'One', conf: 1 },
+  { type: 'word', source: 'source-2', sourceStart: 3, length: 1, word: 'Two', conf: 1 },
+  { type: 'word', source: 'source-2', sourceStart: 4, length: 1, word: 'Three', conf: 1 },
+  { type: 'word', source: 'source-2', sourceStart: 5, length: 1, word: 'Four', conf: 1 },
+  { type: 'artificial_silence', length: 10 },
+];
+
+beforeAll(async () => {
+  await page.waitForSelector('h1');
+
+  const document: Document = {
+    content: testContent,
+    sources: {
+      'source-1': {
+        objectUrl: 'blob://source-1',
+        fileContents: new ArrayBuffer(0),
+      },
+      'source-2': {
+        objectUrl: 'blob://source-2',
+        fileContents: new ArrayBuffer(0),
+      },
+    },
+  };
+
+  await page.evaluate((document) => {
+    window.store.dispatch(window.reducers.openDocumentFromMemory(document));
+  }, document as unknown as JSONObject);
+});
+
+test('Document renders item 13', async () => {
+  await page.waitForSelector('#item-13');
+});
+
+async function getState(): Promise<RootState> {
+  return (await page.evaluate(() => window.store.getState())) as unknown as RootState;
+}
+
+async function moveMouseToElement(selector: string) {
+  const item = await page.waitForSelector(selector);
+  assertSome(item);
+  const bb = await item.boundingBox();
+  assertSome(bb);
+  await page.mouse.move(bb.x + bb.width / 2, bb.y + bb.height / 2);
```

Comment:
Placing this exact in the middle of the element seems dangerous, as it could lead to flaky tests (rounding-error sometimes placing it a bit off). Maybe we could move the cursor a bit more to one side?